test/specs/element.rectangle.tests.js

```diff
@@ -264,35 +264,112 @@ describe('Rectangle element tests', function() {
 		rectangle.draw();
 
 		expect(mockContext.getCalls()).toEqual([{
-			name: 'beginPath',
-			args: [],
-		}, {
 			name: 'setFillStyle',
 			args: ['rgb(255, 0, 0)']
 		}, {
 			name: 'setStrokeStyle',
 			args: ['rgb(0, 0, 255)'],
 		}, {
-			name: 'setLineWidth',
-			args: [undefined]
-		}, {
-			name: 'moveTo',
-			args: [8, 0]
-		}, {
-			name: 'lineTo',
-			args: [8, 15]
-		}, {
-			name: 'lineTo',
-			args: [12, 15]
-		}, {
-			name: 'lineTo',
-			args: [12, 0]
-		}, {
-			name: 'fill',
-			args: [],
+			name: 'fillRect',
+			args: [8, 0, 4, 15],
 		}]);
 	});
 
+	it('should draw borders correctly when horizontal and pointing right', function() {
+		var mockContext = window.createMockContext();
+		var rectangle = new Chart.elements.Rectangle({
+			_chart: {ctx: mockContext}
+		});
+
+		// Attach a view object as if we were the controller
+		rectangle._view = {
+			horizontal: true,
+			borderWidth: 2,
+			borderSkipped: false,
+			ctx: mockContext,
+			base: 0,
+			height: 5,
+			x: 15,
+			y: 10,
+		};
+
+		rectangle.draw();
+
+		var drawCalls = rectangle._view.ctx.getCalls().slice(4);
+		expect(drawCalls).toEqual([
```

Comment:
I think we really should prefer image based test instead of checking draw calls.

Reply:
Actually, [these image tests](3988) have been introduced to avoid doing this:

> Attempt to make easier the creation of unit tests that check the drawing output. Until now, this was done by checking calls on a 'fake' context, which is hard to maintain (need to update pixel values by hands) and also not reliable when optimizing code (i.e. different calls sequence but same result).

Reply:
These can make sense, if its known to be optimal to draw in a certain pattern. (IDK if it is)
It was also faster to do these - but I can add / change to image based if everything else is ok.

Reply:
Do we still want these tests if we have the image-based ones?

Reply:
The image-based tests look good, and I think the mock-based tests are no longer necessary.

Reply:
Removed mock tests and altered couple of fixtures to catch new things